### ISSUE TITLE
Migrate to from java.util.Date to the new JDK 8 classes.

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#1541]
# Task Description

As suggested by @ptirador , we should consider migrating from `java.util.Date` to the new date classes in JDK8.

# Tasks

The following tasks will need to be carried out:
* [ ] Investigate, if this will affect things for OrientDB.
* [ ] Migrate the code to the new date classes.

# Useful Links

* [Baeldung: Migrating to Java 8 Date/Time API](https://www.baeldung.com/migrating-to-java-8-date-time-api)

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
 * @ptirador 
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

**Comment by fuss86** [comment]
It will not work with `orientDb` out-of-the box. See https://orientdb.com/docs/last/Types.html

**Comment by carlspring** [comment]
Thanks for pointing this out, @fuss86 !

@ptirador , thanks for the suggestion, but I guess this will have to be put on hold, until OrientDB decide to do something about it. :(

**Comment by ptirador** [comment]
@carlspring: No problem about it.
@fuss86: Thanks for clarifying :)

**Comment by mslowiak** [comment]
FYI: https://github.com/orientechnologies/orientdb/issues/3652

**Comment by ptirador** [comment]
Thanks for the info, @mslowiak! Let's vote it up!

**Comment by carlspring** [comment]
@sbespalov :

Would this make sense after #1649 ( #1676 ), or has it already been looked at? I believe I noticed you're already using this for the entities in Janusgraph/Cassandra.

Should we close this task, or keep it as a follow-up of #1649 ?

**Comment by sbespalov** [comment]
we should keep it

**Comment by carlspring** [comment]
Okay, I'll leave it open.
